app/src/main/java/science/apolline/service/sensor/BluetoothLeService.kt

```diff
@@ -0,0 +1,620 @@
+                        try {
+                            output = FileOutputStream(myFile, true)
+                        } catch (e: FileNotFoundException) {
+                            e.printStackTrace()
+                        }
+
+                        try {
+                            output?.write(data.toByteArray())
+                        } catch (e: IOException) {
+                            e.printStackTrace()
+                        }
+
+                        try {
+                            output?.close()
+                        } catch (e: IOException) {
+                            e.printStackTrace()
+                        }
+
+                    } else {
+                        Log.e("écriture fichier", "ERROR DE CREATION DE DOSSIER")
+                    }
+
+                    // récupérer info GPS android
+                    val locationManager = getSystemService(Context.LOCATION_SERVICE) as LocationManager
+                    val criteria = Criteria()
+                    if (checkSelfPermission(this@BluetoothLeService, Manifest.permission.ACCESS_FINE_LOCATION) !== PackageManager.PERMISSION_GRANTED && checkSelfPermission(this@BluetoothLeService, Manifest.permission.ACCESS_COARSE_LOCATION) !== PackageManager.PERMISSION_GRANTED) {
+                        return
+                    }
+                    var location: Location? = null
+                    if (locationManager != null) {
+                        location = locationManager.getLastKnownLocation(locationManager.getBestProvider(criteria, false))
+                    }
+                    if (location != null) {
+                        Lat = location.latitude
+                        Long = location.longitude
+                    }
+
+
+
+
+
+
+
+                    var ioiofm : android.support.v4.app.Fragment? = MainActivity.mFragment
+
+                    ioiofm!!.fragment_ioio_progress_pm1.setValueAnimated(MainActivity.mBluetoothLeService!!.appaData.pm01Value.toFloat())
+                    ioiofm.fragment_ioio_progress_pm2_5.setValueAnimated(MainActivity.mBluetoothLeService!!.appaData.pm2_5Value.toFloat())
+                    ioiofm.fragment_ioio_progress_pm10.setValueAnimated(MainActivity.mBluetoothLeService!!.appaData.pm10Value.toFloat())
+                    ioiofm.fragment_ioio_progress_rht.setValueAnimated(MainActivity.mBluetoothLeService!!.appaData.humi.toFloat())
+                    ioiofm.fragment_ioio_progress_tmpk.setValueAnimated(MainActivity.mBluetoothLeService!!.appaData.tempKelvin)
+                    ioiofm.fragment_ioio_progress_tmpc.setValueAnimated(MainActivity.mBluetoothLeService!!.appaData.tempe.toFloat())
+                    ioiofm.fragment_ioio_progress_tmpk.setValueAnimated(MainActivity.mBluetoothLeService!!.appaData.tempe.toFloat()+273.15.toFloat())
+
+                   // println("tempppp "  + (MainActivity.img == null))
```

Comment:
remove if useless

Reply:
You should use Log instead of print and surround it with a config check debug if only needed for debug